### DISCUSSION TITLE
Git folder's not created

### DISCUSSION THREAD

**Discussion by ihkim8** [community/community#41796] [New to GitHub]
Hihi
How do you create git folder? 
As far as I understood, it should be created by itself as I use terminal. 
I had several cases: 1. git folder was created but did not appear on vs code 2. git folder was not created.
However, for both cases there were times where it was uploaded on github! 
I just have no clue how to create git folder and make it shown on vs code. 

Cheers

**Comment by radar-blip** [comment]
You cannot create an empty folder, there must be something in it for the folder to be created.
Since you are using the terminal, maybe this will help:
https://careerkarma.com/blog/git-create-folder-in-github/

**Comment by sadman-shami** [comment]
To show .git folder in VSCode put the following lines in vscode settings

```
  "files.exclude": {
    "**/.git": false,
  },
```

**Comment by igorskyflyer** [comment]
Your question is not very clear and it can mean several things so I'll try to cover all cases:
- .git folder is used by Git/GitHub to store the actual data it uses, indices, changes, etc., you create it by running `git init` in a terminal inside of your work folder, this folder should NOT be uploaded to GitHub manually,
- to create a regular folder in your work folder from the GitHub site, you can't create empty folders, unless you create an empty file in it, e.g. create a file named `new-folder/empty.file` and the folder will be created, for this folder to appear in your local work folder you need to do a `git pull`,
- to create a folder inside your local  work folder in VS Code, just create a regular folder but for it to appear on GitHub you need to do add it to Git, commit it and then do a `git push`.

To summarize, you need `git pull` when creating files/folders on GitHub and you want them to appear in your local/VS Code work folder and `git push` when you want local/VS Code files/folders to appear on GitHub. Hope this helps.

I recommend going through Git's [Official documentation](https://git-scm.com/docs/gittutorial) for proper use cases.

**Comment by swi-1** [comment]
Hello , The .git folder is created by default when you do a "git init". That's why you can see it by default @sadman-shami explained that on your settings vs code you can make the hidden folders appear in vscode.
